src/tests/krate.rs

```diff
@@ -1682,23 +1716,33 @@ fn prerelease_versions_not_included_in_reverse_dependencies() {
 fn yanked_versions_not_included_in_reverse_dependencies() {
     let (_b, app, middle) = ::app();
 
-    let v100 = semver::Version::parse("1.0.0").unwrap();
-    let v200 = semver::Version::parse("2.0.0").unwrap();
-    let mut req = ::req(app, Method::Get, "/api/v1/crates/c1/reverse_dependencies");
-    ::mock_user(&mut req, ::user("foo"));
-    let (c1, _) = ::mock_crate_vers(&mut req, ::krate("c1"), &v100);
-    let _ = ::mock_crate_vers(&mut req, ::krate("c2"), &v100);
-    let (_, c2v2) = ::mock_crate_vers(&mut req, ::krate("c2"), &v200);
-
-    ::mock_dep(&mut req, &c2v2, &c1, None);
+    let mut req = ::req(
+        app.clone(),
+        Method::Get,
+        "/api/v1/crates/c1/reverse_dependencies",
+    );
+    {
+        let conn = app.diesel_database.get().unwrap();
+        let u = ::new_user("foo").create_or_update(&conn).unwrap();
+        let c1 = ::CrateBuilder::new("c1", u.id)
+            .version("1.0.0")
+            .expect_build(&conn);
+        ::CrateBuilder::new("c2", u.id)
+            .version("1.0.0")
+            .version(::VersionBuilder::new("2.0.0").dependency(&c1, None))
+            .expect_build(&conn);
+    }
 
     let mut response = ok_resp!(middle.call(&mut req));
     let deps = ::json::<RevDeps>(&mut response);
     assert_eq!(deps.dependencies.len(), 1);
     assert_eq!(deps.meta.total, 1);
     assert_eq!(deps.dependencies[0].crate_id, "c2");
 
-    t!(c2v2.yank(req.tx().unwrap(), true));
+    diesel::update(versions::table.filter(versions::num.eq("2.0.0")))
```

Comment:
AFAIK this part could be changed back to calling `version.yank()` again once yank is converted to use diesel; I'm going to add in a note here because I'd prefer to have it call the method in case we add extra behavior to the yank method someday.